I'm seeing an ICE with gcc-8.2.0 optimized:
```[ 89%] Building CXX object tests/Unit/PointwiseFunctions/MathFunctions/CMakeFiles/Test_MathFunctions.dir/Test_TensorProduct.cpp.o
In file included from /home/wthrowe/spectre/tests/Unit/PointwiseFunctions/MathFunctions/Test_TensorProduct.cpp:16:
/home/wthrowe/spectre/src/DataStructures/Variables.hpp: In substitution of ‘template<class ... WrappedTags, typename Requires_detail::requires_impl<flat_all_v<is_same_v<typename db::databox_detail::remove_all_prefixes_impl<WrappedTags, is_base_of_v<db::PrefixTag, WrappedTags> >::type, typename db::databox_detail::remove_all_prefixes_impl<Tags, is_base_of_v<db::PrefixTag, Tags> >::type>...> >::template_error_type_failed_to_meet_requirements_on_template_parameters <anonymous> > Variables<brigand::list<{anonymous}::Var1, {anonymous}::Var2<1> > >::Variables(const Variables<brigand::list<Ts ...> >&) [with WrappedTags = {{anonymous}::Var1, {anonymous}::Var2<1>}; typename Requires_detail::requires_impl<flat_all_v<is_same_v<typename db::databox_detail::remove_all_prefixes_impl<WrappedTags, is_base_of_v<db::PrefixTag, WrappedTags> >::type, typename db::databox_detail::remove_all_prefixes_impl<Tags, is_base_of_v<db::PrefixTag, Tags> >::type>...> >::template_error_type_failed_to_meet_requirements_on_template_parameters <anonymous> = <missing>]’:
/home/wthrowe/spectre/src/NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp:39:14:   required from ‘Variables<typename brigand::detail::transform<0, DerivativeTags, brigand::bind<Tags::deriv, brigand::_1, brigand::pin<brigand::integral_constant<long unsigned int, Dim> >, brigand::pin<DerivativeFrame> > >::type> partial_derivatives(const Variables<TagList>&, const Mesh<Dim>&, InverseJacobian<DataVector, Dim, Frame::Logical, DerivativeFrame>&) [with DerivativeTags = brigand::list<{anonymous}::Var1, {anonymous}::Var2<1> >; VariableTags = brigand::list<{anonymous}::Var1, {anonymous}::Var2<1> >; long unsigned int Dim = 1; DerivativeFrame = Frame::Inertial; typename brigand::detail::transform<0, DerivativeTags, brigand::bind<Tags::deriv, brigand::_1, brigand::pin<brigand::integral_constant<long unsigned int, Dim> >, brigand::pin<DerivativeFrame> > >::type = brigand::list<Tags::deriv<{anonymous}::Var1, brigand::integral_constant<long unsigned int, 1>, Frame::Inertial>, Tags::deriv<{anonymous}::Var2<1>, brigand::integral_constant<long unsigned int, 1>, Frame::Inertial> >; InverseJacobian<DataVector, Dim, Frame::Logical, DerivativeFrame> = Tensor<DataVector, brigand::list<brigand::integral_constant<int, 2>, brigand::integral_constant<int, 1> >, brigand::list<Tensor_detail::TensorIndexType<1, (UpLo)0, Frame::Logical, (IndexType)0>, Tensor_detail::TensorIndexType<1, (UpLo)1, Frame::Inertial, (IndexType)0> > >]’
/home/wthrowe/spectre/tests/Unit/PointwiseFunctions/MathFunctions/Test_TensorProduct.cpp:182:46:   required from ‘void {anonymous}::test_with_numerical_derivatives(const Mesh<Dim>&, double, std::array<std::unique_ptr<MathFunction<1> >, Dim>&&) [with long unsigned int VolumeDim = 1]’
/home/wthrowe/spectre/tests/Unit/PointwiseFunctions/MathFunctions/Test_TensorProduct.cpp:235:31:   required from here
/home/wthrowe/spectre/src/DataStructures/Variables.hpp:126:22: internal compiler error: in tsubst_pack_expansion, at cp/pt.c:12200
             Requires<tmpl2::flat_all_v<
                      ^~~~~
0x592bf0 tsubst_pack_expansion(tree_node*, tree_node*, int, tree_node*)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:12200
0x6c49fa tsubst_template_args
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:12433
0x6cbb38 tsubst_copy_and_build(tree_node*, tree_node*, int, tree_node*, bool, bool)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:17786
0x6bf831 tsubst_expr(tree_node*, tree_node*, int, tree_node*, bool)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:17434
0x6c11e3 tsubst_expr(tree_node*, tree_node*, int, tree_node*, bool)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:11427
0x6c11e3 tsubst_template_arg
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:11438
0x6c49ca tsubst_template_args
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:12462
0x6c7856 tsubst_aggr_type
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:12664
0x6bb213 tsubst(tree_node*, tree_node*, int, tree_node*)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:14825
0x6ced00 convert_template_argument
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:7998
0x6d5217 type_unification_real
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:20475
0x6d5c5b fn_type_unification(tree_node*, tree_node*, tree_node*, tree_node* const*, unsigned int, tree_node*, unification_kind_t, int, bool, bool)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/pt.c:19619
0x5d2357 add_template_candidate_real
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/call.c:3179
0x5cf483 add_template_candidate
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/call.c:3258
0x5cf483 add_candidates
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/call.c:5520
0x5d6dd5 add_candidates
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/call.c:5444
0x5d6dd5 build_new_method_call_1
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/call.c:9245
0x5d6dd5 build_new_method_call(tree_node*, tree_node*, vec<tree_node*, va_gc, vl_embed>**, tree_node*, int, tree_node**, int)
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/call.c:9453
0x661565 locate_fn_flags
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/method.c:1024
0x661b6f walk_field_subobs
        /var/tmp/portage/sys-devel/gcc-8.2.0/work/gcc-8.2.0/gcc/cp/method.c:1439
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://bugs.gentoo.org/> for instructions.
make[4]: *** [tests/Unit/PointwiseFunctions/MathFunctions/CMakeFiles/Test_MathFunctions.dir/build.make:143: tests/Unit/PointwiseFunctions/MathFunctions/CMakeFiles/Test_MathFunctions.dir/Test_TensorProduct.cpp.o] Error 1
```

--- a/src/DataStructures/Variables.hpp
+++ b/src/DataStructures/Variables.hpp
@@ -123,9 +123,9 @@ class Variables<tmpl::list<Tags...>> {
   // @{
   /// Copy and move semantics for wrapped variables
   template <typename... WrappedTags,
-            Requires<tmpl2::flat_all_v<
-                cpp17::is_same_v<db::remove_all_prefixes<WrappedTags>,
-                                 db::remove_all_prefixes<Tags>>...>> = nullptr>
+            Requires<tmpl2::flat_all_v<std::is_same<
+                db::remove_all_prefixes<WrappedTags>,
+                db::remove_all_prefixes<Tags>>::value...>> = nullptr>
   explicit Variables(Variables<tmpl::list<WrappedTags...>>&& rhs) noexcept;
   template <typename... WrappedTags,
             Requires<tmpl2::flat_all_v<
@@ -474,9 +474,10 @@ Variables<tmpl::list<Tags...>>& Variables<tmpl::list<Tags...>>::operator=(
 }
 
 template <typename... Tags>
-template <typename... WrappedTags, Requires<tmpl2::flat_all_v<cpp17::is_same_v<
-                                       db::remove_all_prefixes<WrappedTags>,
-                                       db::remove_all_prefixes<Tags>>...>>>
+template <typename... WrappedTags,
+          Requires<tmpl2::flat_all_v<
+              std::is_same<db::remove_all_prefixes<WrappedTags>,
+                           db::remove_all_prefixes<Tags>>::value...>>>
 Variables<tmpl::list<Tags...>>::Variables(
     Variables<tmpl::list<WrappedTags...>>&& rhs) noexcept
     : variable_data_impl_(std::move(rhs.variable_data_impl_)),

--- a/src/Domain/CoordinateMaps/CoordinateMap.hpp
+++ b/src/Domain/CoordinateMaps/CoordinateMap.hpp
@@ -286,7 +286,7 @@ class CoordinateMap
   // clang-tidy: google-runtime-references
   void pup(PUP::er& p) override {  // NOLINT
     CoordinateMapBase<SourceFrame, TargetFrame, dim>::pup(p);
-    p | maps_;
+    PUP::pup(p, maps_);
   }
 
  private:

--- a/src/Evolution/Conservative/Tags.hpp
+++ b/src/Evolution/Conservative/Tags.hpp
@@ -43,8 +43,9 @@ struct ComputeNormalDotFlux : db::add_tag_prefix<NormalDotFlux, Tag>,
     auto result = make_with_value<
         ::Variables<db::wrap_tags_in<NormalDotFlux, tags_list>>>(flux, 0.);
 
-    tmpl::for_each<tags_list>([&result, &flux, &normal](auto tag) noexcept {
-      using tensor_tag = tmpl::type_from<decltype(tag)>;
+    tmpl::for_each<tags_list>([&result, &flux,
+                               &normal ](auto local_tag) noexcept {
+      using tensor_tag = tmpl::type_from<decltype(local_tag)>;
       auto& result_tensor = get<NormalDotFlux<tensor_tag>>(result);
       const auto& flux_tensor =
           get<Flux<tensor_tag, tmpl::size_t<VolumeDim>, Fr>>(flux);

--- a/tests/Unit/DataStructures/DataBox/Test_Deferred.cpp
+++ b/tests/Unit/DataStructures/DataBox/Test_Deferred.cpp
@@ -322,6 +322,6 @@ SPECTRE_TEST_CASE("Unit.DataStructures.DataBox.Deferred.PupNonfunction",
   ERROR_TEST();
   Deferred<double> deferred{3.89};
   auto data = std::make_unique<char[]>(10);
-  PUP::fromMem p{static_cast<const void* const>(data.get())};
+  PUP::fromMem p{static_cast<const void*>(data.get())};
   deferred.pack_unpack_lazy_function(p);
 }

--- a/tests/Unit/Domain/Test_ElementIndex.cpp
+++ b/tests/Unit/Domain/Test_ElementIndex.cpp
@@ -28,7 +28,14 @@ void check(const size_t block1,
   ElementIndex<VolumeDim> element_index1{}, element_index2{};
   // Check for nondeterminacy due to previous memory state.
   std::memset(&element_index1, 0, sizeof(element_index1));
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ > 7)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif  // defined(__GNUC__) && !defined(__clang__) && (__GNUC__ > 7)
   std::memset(&element_index2, 255, sizeof(element_index2));
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ > 7)
+#pragma GCC diagnostic pop
+#endif  // defined(__GNUC__) && !defined(__clang__) && (__GNUC__ > 7)
   new (&element_index1) ElementIndex<VolumeDim>(id1);
   new (&element_index2) ElementIndex<VolumeDim>(id2);
 

